cms/djangoapps/fdcc_utils/tasks.py

```diff
@@ -0,0 +1,91 @@
+from datetime import date, timedelta
+import logging
+import os
+import six
+
+from celery.task import task
+from celery_utils.logged_task import LoggedTask
+from django.contrib.auth.models import User
+from django.core.management.base import CommandError
+
+from opaque_keys import InvalidKeyError
+from opaque_keys.edx.keys import CourseKey
+from xmodule.modulestore.django import modulestore
+from contentstore.views.certificates import CertificateManager, _get_course_and_check_access
+from contentstore.views.item import _get_xblock, _save_xblock
+from fdcc_utils.utils import get_course, get_sections
+from lms.djangoapps.certificates.api import set_cert_generation_enabled
+
+log = logging.getLogger(__name__)
+
+
+@task(base=LoggedTask)
+def configure_fdcc_certificate(course_id=None, cert_is_active=None, self_generation_enabled=None):
+    log.info(u"Configuring Certificate for course %s - cert_is_active: %s - self_generation_enabled: %s",
+             course_id, cert_is_active, self_generation_enabled)
+
+    if not cert_is_active and self_generation_enabled is True:
+        raise CommandError(("Enabling certificate self generation without activating the certificate "
+                            "will not make the certificate available to students."))
+
+    try:
+        course_key = CourseKey.from_string(course_id)
+    except InvalidKeyError:
+        raise CommandError(u"Invalid course_key: '%s'." % course_id)
+
+    store = modulestore()
+    if not store.get_course(course_key):
+        raise CommandError(u"Course with %s key not found." % course_id)
+
+    # De-/activate Certificate
+    fdcc_admin = User.objects.get(username=os.getenv('FDCC_ADMIN_USERNAME'))
```

Comment:
The admin thinkg seems odd, insofar as it's mixing runtime information with deploy time information

Reply:
As mentioned in our call the open-edx functions that are used require a user for traceability in the CMS data